ProcessMaker/EncryptedData/Vault.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace ProcessMaker\EncryptedData;
+
+use GuzzleHttp\Client;
+use ProcessMaker\EncryptedData\EncryptedDataInterface;
+use ProcessMaker\Models\EncryptedData;
+use VaultPHP\Authentication\Provider\Token;
+use VaultPHP\SecretEngines\Engines\Transit\Request\DecryptData\DecryptDataRequest;
+use VaultPHP\SecretEngines\Engines\Transit\Request\EncryptData\EncryptDataRequest;
+use VaultPHP\SecretEngines\Engines\Transit\Transit;
+use VaultPHP\VaultClient;
+
+class Vault implements EncryptedDataInterface
+{
+    public static $iv = '';
+
+    private $host;
+    private $token;
+    private $transitKey;
+
+    public function __construct()
+    {
+        // Get config
+        $this->host = config('app.encrypted_data.vault_host');
+        $this->token = config('app.encrypted_data.vault_token');
+        $this->transitKey = config('app.encrypted_data.vault_transit_key');
+    }
+
+    /**
+     * Encrypt text.
```

Comment:
Please complete the documentation, explain when the $iv and $key can be optional, when not.